control-base/src/main/java/fi/nls/oskari/util/WFSGetLayerFields.java

```diff
@@ -0,0 +1,157 @@
+package fi.nls.oskari.util;
+
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.service.ServiceException;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+public class WFSGetLayerFields {
+    private static final String STRING = "string";
+    private static final String NUMBER = "number";
+    private static final String BOOLEAN = "boolean";
+    private static final String UNKNOWN = "unknown";
+    private static final String ATTRIBUTES_KEY = "attributes";
+    private static final String GEOMETRY_FIELD_KEY = "geometryField";
+    /**
+     * Return fields information for the WFS layer
+     *
+     * The result is constructed as:
+     * {
+     *     "attributes": {
+     *         "field-1": STRING,
+     *         "field-2": NUMBER,
+     *         ...
+     *         "field-n": BOOLEAN
+     *     },
+     *     "geometryField": "geometry"
+     * }
+     *
+     * The field type can be one of the following values:
+     *  - string
+     *  - number
+     *  - boolean
+     *  - unknown
+     *
+     */
+    public static JSONObject getLayerFields(OskariLayer layer) throws ServiceException {
+        return layer.getVersion().startsWith("3") ? getCollectionFields(layer) : getFeatureTypeFields(layer);
+    }
+
+    /**
+     * Return fields information for WFS 3.x layer
+     *
+     * The fields are collected by requesting 10 features from the
+     * collection and derive the attribute type from the attribute
+     * value, which is either NUMBER or STRING. If the attribute
+     * type cannot be derived from any feature, e.g. all 10 features
+     * have null values on the attribute, then the attribute will
+     * have an "unknown" type
+     *
+     */
+    private static JSONObject getCollectionFields(OskariLayer layer) throws ServiceException {
+        String url = layer.getUrl() + "collections/" + layer.getName() + "/items";
```

Comment:
Use https://github.com/oskariorg/oskari-server/blob/master/service-wfs3/src/main/java/org/oskari/service/wfs3/OskariWFS3Client.java#L229 (make the static method public on the client) so we see where it's used. I don't think we should use the existing client for getting the features since it's probably doing too much for what we need here. We could use `OskariWFS3Client.validateResponse(conn, CONTENT_TYPE_GEOJSON);` though for sanity checking the response.